libs/oauth-key-service/src/main/java/uk/gov/di/ipv/core/library/oauthkeyservice/OAuthKeyService.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.di.ipv.core.library.oauthkeyservice;
+
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.nimbusds.jose.jwk.RSAKey;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.tracing.Tracing;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.dto.OauthCriConfig;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.ParseException;
+import java.util.List;
+
+import static com.nimbusds.oauth2.sdk.http.HTTPResponse.SC_OK;
+
+public class OAuthKeyService {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final HttpClient httpClient;
+
+    @ExcludeFromGeneratedCoverageReport
+    public OAuthKeyService() {
+        this.httpClient = HttpClient.newHttpClient();
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public OAuthKeyService(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public RSAKey getValidEncryptionKey(OauthCriConfig criConfig) throws ParseException {
```

Comment:
What does `valid` mean here?

Reply:
hmm I think I was trying to refer to the fact we have a bit of filtering ie the first key that has `use:enc` but I can see how this doesn't exactly imply `valid` either. How about just `getEncryptionKey?`

Reply:
Sounds good to me